t/ConfigStore/Base.t

```diff
@@ -24,12 +24,153 @@ BEGIN {
 
+is_deeply([], [$configStore->search()], "Search for nothing get nothing");
+
+is_deeply(
+    [
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+        },
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+        },
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+        },
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+        }
+    ],
+    [ $configStore->search( "param1", "value1", ) ],
+    "Search param1=value1"
+);
+
+is_deeply(
+    [
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+            'id'     => 'section1'
+        },
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+            'id'     => 'section1 group 1',
+        },
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+            'id'     => 'section1 group 2',
+        },
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+            'id'     => 'section2',
+        }
+    ],
+    [ $configStore->search( "param1", "value1", 'id' ) ],
+    "Search param1=value1 with idKey=id"
+);
+
+is_deeply(
+    [
+        {
+            'param3' => 'value3',
+            'param4' => 'value4',
+            'id'     => 'default'
+        },
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+            'id'     => 'section1'
+        },
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+            'id'     => 'section1 group 1',
+        },
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+            'id'     => 'section1 group 2',
+        },
+        {
+            'param1' => 'value1',
+            'param2' => 'value2',
+            'param3' => 'value3',
+            'param4' => 'value4',
+            'id'     => 'section2',
+        }
+    ],
+    [ $configStore->search( "param3", "value3", 'id' ) ],
+    "Search inhertited values param3=value3 with idKey=id"
+);
+
+is_deeply([], [$configStore->search_like()], "Search like for nothing get nothing");
```

Comment:
Not sure I get that sentence.

Sounds like Confucius

![image](https://user-images.githubusercontent.com/3857942/34046113-de75d884-e179-11e7-846f-ec12557971fe.png)